Multi-gpu code 추가 (develop branch)

lilly9117 opened the pull request:
multi-gpu 사용 위해 데이터 병렬 처리 코드 추가하였습니다

lilly9117 committed:
add multi-gpu code

mwmw7 commented:
```
  CUDA_VISIBLE_DEVICES=1 python3 src/train.py --dataset TU-Berlin --dim-out 64 --semantic-models word2vec-google-news --epochs 1000 --early-stop 200 --lr 0.0001 --batch-size 200 --ngpu 2
Parameters:     Namespace(batch_size=200, dataset='TU-Berlin', dim_out=64, early_stop=200, epoch_size=100, epochs=1000, filter_sketch=False, gamma=0.1, gzs_sbir=False, im_sz=224, lambda_disc_im=0.5, lambda_disc_se=0.25, lambda_disc_sk=0.5, lambda_gen_adv=1.0, lambda_gen_cls=1.0, lambda_gen_cyc=1.0, lambda_gen_reg=0.1, lambda_im=10.0, lambda_regular=0.001, lambda_se=10.0, lambda_sk=10.0, log_interval=1, lr=0.0001, milestones=[], momentum=0.9, ngpu=2, num_workers=4, number_qualit_results=200, save_best_results=False, save_image_results=False, semantic_models=['word2vec-google-news'], sk_sz=224, split_eccv_2018=False, test=False)
Checkpoint path: /ml_data/sem_pcyc/aux/CheckPoints/TU-Berlin/word2vec-google-news/64
Logger path: /ml_data/sem_pcyc/aux/LogFiles/TU-Berlin/word2vec-google-news/64
Result path: /ml_data/sem_pcyc/aux/Results/TU-Berlin/word2vec-google-news/64
Loading data...Done
Initializing model variables...Done
Initializing trainable models...Done
Defining optimizers...Done
Defining losses...Done
Initializing variables...Done
Setting logger...Done
Checking cuda...*Cuda exists*...Done
***Train***
Traceback (most recent call last):
  File "src/train.py", line 364, in <module>
    main()
  File "src/train.py", line 228, in main
    sem_pcyc_model.scheduler_gen.step()
  File "/home/mwmw7/anaconda3/envs/etri/lib/python3.6/site-packages/torch/nn/modules/module.py", line 576, in __getattr__
    type(self).__name__, name))
AttributeError: 'DataParallel' object has no attribute 'scheduler_gen'                                                                                                                                                                                                                                                                                                                                                                                                                      exit:1
    i) 

```

lilly9117 committed:
dataparallel error 해결

lilly9117 commented:
위 error 해결해서 다시 push 했습니다 @mwmw7

mwmw7 committed:
Add nvidia-smi 설명

mwmw7 committed:
Add comment in test.py
Update config.ini for run in hosup's machine

mwmw7 committed:
Add multigpu selection to train or test

mwmw7 committed:
Add print to debug
Update *.jpg -> *
Add intersection dataset

mwmw7 committed:
Add runScript

mwmw7 committed:
Remove unused TU-Berlin dataset

mwmw7 committed:
Update train_intersection_data_word2vec_batch_32.run.xml

mwmw7 committed:
load * files because of .base64

mwmw7 committed:
Update README.md

mwmw7 committed:
Remove temporary to train

mwmw7 committed:
Add gc
Add cache remove

mwmw7 committed:
Update README.md

mwmw7 committed:
Fix glob */* -> */.base64 to read tooning thumbnail

mwmw7 committed:
Update README.md

mwmw7 committed:
Reformat README.md

mwmw7 committed:
Add intersection for test.py

mwmw7 committed:
Add runscript for test and train

mwmw7 committed:
Update README.md
Add condition for loading intersection
Add runScript
Add gc.collect()
Add empty_cache()

mwmw7 committed:
Add runScript

mwmw7 committed:
Add print code

mwmw7 committed:
Add runscript

mwmw7 committed:
Add baseHandler and sem_pcyc_handler.py

mwmw7 committed:
Add .gitignore

mwmw7 committed:
Add init test code

mwmw7 committed:
Remove import util from model.py
Add config.properties

mwmw7 committed:
Add run scrip to make .mar and run torchserve using docker
r

mwmw7 committed:
Fix model and handler to road weight for SEM_PCYC

mwmw7 committed:
Add parms to init SEM_PCYC

mwmw7 committed:
Fix sketch typo and Success model init

mwmw7 committed:
Add update model archiving script

mwmw7 committed:
Update orchserve config.properties

kimyw97 committed:
sem_pcyc handler custom

kimyw97 committed:
mk_image_emd_npy.py for saving embedded images to numpy

kimyw97 committed:
branden's sem_pcyc_handler clone

DongHyukShin93 committed:
make npy applied handler.py

kimyw97 committed:
Apply image_embedding.npy to mar file

kimyw97 committed:
File(Class version) to create image_embedding.npy

kimyw97 committed:
model.py custom by paul

kimyw97 committed:
Add DataGeneratorImage class to model.py, custom handler

kimyw97 committed:
Update README.md

Manual for deploying services using torchserve

kimyw97 committed:
Update README.md

kimyw97 committed:
Update README.md

DongHyukShin93 committed:
make image embedding npy file update

DongHyukShin93 committed:
make image embedding npy file update

DongHyukShin93 committed:
Merge remote-tracking branch 'origin/nipa_paul' into nipa_paul

DongHyukShin93 committed:
make image embedding npy file update

DongHyukShin93 committed:
git pull test

monkeykim111 committed:
push test

DongHyukShin93 committed:
git pull test

DongHyukShin93 committed:
Merge remote-tracking branch 'origin/nipa_paul' into nipa_paul

DongHyukShin93 committed:
argsort(-inference_output) add "-"

kimyw97 committed:
mk_image_emd_npy.py trouble shooting

kimyw97 committed:
Test case 2942 using "images_embedding.npy" success

monkeykim111 committed:
Update README.md

kimyw97 committed:
Update README.md

kimyw97 committed:
Update README.md

kimyw97 committed:
Update README.md

kimyw97 committed:
Update README.md

kimyw97 committed:
Update README.md

kimyw97 committed:
Update README.md

kimyw97 committed:
fixed function name(load_files_trberlin_zeroshot)
using "np.concatenate" fuction instead "+"

kimyw97 committed:
Merge remote-tracking branch 'origin/nipa_paul' into nipa_paul

Commit:
new class word2vec function

Commit:
merge develop

Commit:
Merge branch 'nipa_paul' of https://github.com/toonsquare/sem-pcyc into develop

 Conflicts:
	README.md
	config.ini
	src/models.py
	src/test.py
	src/train.py
